### PR TITLE
[dev2][conan list] Enhancements 2.0

### PR DESCRIPTION
**Pull request by franramirez688**
Changelog: Bugfix: Ordering if same ref.name but different versions.
Changelog: Feature: Avoid requesting package configuration if PkgID is passed.
Docs: omit

Related to https://github.com/conan-io/conan/pull/12798

### CODE REVIEW

File: conan/cli/commands/list.py [hunk 1]
```diff
@@ -20,57 +20,57 @@
...
-                for pref, binary_info in prefs:
-                    pref_date = f" ({timestamp_to_str(pref.timestamp)})" if pref.timestamp else ""
-                    if search_mode == ListPatternMode.SHOW_PACKAGE_IDS:
-                        cli_out_write(f"{indentation * 3}PID: {pref.package_id}{pref_date}",
-                                      fg=reference_color)
-                        if not binary_info:
-                            cli_out_write(f"{indentation * 4}Empty package information",
+        for ref_name, refs in info_per_ref_name.items():
+            cli_out_write(f"{indentation}{ref_name}", fg=recipe_name_color)
+            for ref, prefs in refs.items():
```
Comment (franramirez688) [inline conan/cli/commands/list.py:38]: From here to below, there are no changes at all. It's only the indentation diff.

File: conan/cli/commands/list.py [hunk 2]
```diff
@@ -114,7 +114,7 @@ def list(conan_api: ConanAPI, parser, *args):
             results[name] = {"error": str(e)}
         else:
             results[name] = list_bundle.serialize() if args.format in ("json", "html") \
-                else list_bundle.recipes
+                else list_bundle.ordered_recipes_by_name
```
Comment (memsharded) [inline conan/cli/commands/list.py:117]: I think it is better to leave the ``list_bundle.recipes`` and have that method/property to return the references ordered?

File: conans/test/integration/command_v2/list_test.py [hunk 3]
```diff
@@ -332,6 +345,99 @@ def test_search_package_ids_from_latest_rrev_in_all_remotes_and_cache(self):
...
+                                                                   user="user", channel="channel"))
+        self._add_remote(remote2)
+        self._upload_full_recipe_without_conaninfo(remote2,
+                                                   RecipeReference(name="test_recipe", version="2.1",
+                                                                   user="user", channel="channel"))
+        self.client.run(f'list *#* -r="*" -c')
+        output = str(self.client.out)
+        expected_output = textwrap.dedent("""\
+        Local Cache:
+          test_recipe
```
Comment (memsharded) [inline conans/test/integration/command_v2/list_test.py:365]: Were you ordering recipes by name? ``pkg`` should be before ``test_recipe``?

File: conan/api/model.py [hunk 4]
```diff
@@ -81,15 +81,22 @@ def prefs(self):
...
-        confs = configurations or {}
+        confs = configurations or OrderedDict()
         for pref in prefs:
-            binary_info = confs.get(pref, {})
+            binary_info = confs.get(pref, OrderedDict())
             self.recipes.setdefault(pref.ref, []).append((pref, binary_info))
 
+    @property
+    def ordered_recipes_by_name(self):
+        ret = OrderedDict()
```
Comment (memsharded) [inline conan/api/model.py:91]: This is not a ``SortedDict``, the ``OrderedDict`` is ordered by insertion order, so I think this is not really sorting anything?

**Review by czoido — APPROVED** [review]

**Pull request merged**